packages/tocco-ui/src/EditableValue/typeEditors/DateAbstract.js

```diff
@@ -72,24 +72,23 @@ class DateAbstract extends React.Component {
     return null
   }
 
-  componentWillReceiveProps(props) {
-    this.handleButtonVisibility()
-    const locale = this.getLocale(props.intl.locale)
+  componentDidUpdate(prevProps, prevState, snapshot) {
+    const sameValue = _isEqual(prevProps.value, this.props.value)
+    const sameLocale = _isEqual(prevProps.options.locale, this.props.intl.locale)
+    if (!sameValue || !sameLocale) {
+      const locale = this.getLocale(this.props.intl.locale)
 
-    if (this.Flatpickr && this.flatpickr) {
-      this.Flatpickr.localize(locale)
-      this.flatpickr.set('locale', locale)
+      if (this.Flatpickr && this.flatpickr) {
+        this.Flatpickr.localize(locale)
+        this.flatpickr.set('locale', locale)
 
-      this.flatpickr.set('altFormat', props.options.flatpickrOptions.altFormat)
-      this.flatpickr.setDate(props.value, false)
-      this.flatpickr.redraw()
+        this.flatpickr.set('altFormat', this.props.options.flatpickrOptions.altFormat)
```

Comment:
so if `this.props.options.flatpickrOptions.altFormat` changes this will not update since you wrapped everything in a "only when value or local changed"-if?

...why do you event have this "if"? is the performance not good without?

Reply:
Yes it won't redraw correctly without the condition.